src/pybind/mgr/volumes/fs/nfs.py

```diff
@@ -345,10 +538,10 @@ def delete_export(self, cluster_id, pseudo_path, export_obj=None):
                 self._delete_export_url(common_conf, export.export_id)
                 self.exports[cluster_id].remove(export)
                 self._delete_user(export.fsal.user_id)
+                if not self.exports[cluster_id]:
+                    del self.exports[cluster_id]
                 return 0, "Successfully deleted export", ""
             return 0, "", "Export does not exist"
-        except KeyError:
-            return -errno.EINVAL, "", "Cluster does not exist"
         except Exception as e:
             log.warning("Failed to delete exports")
             return -errno.EINVAL, "", str(e)
```

Comment:
Why are we doing this here and in other methods as well, where we catch all exceptions and return -errno.EINVAL (invalid argument)?

Reply:
e.g., if we want to catch exceptions raised by GaneshaConfParser()'s methods. We can have those methods raise a custon GaneshaError exception and only catch those.

Reply:
All the exceptions don't have error code. So I used `-errno.EINVAL` in the initial code. 
I prefer not throwing entire exception trace back message to user.